### PR TITLE
LanguageDefinition: Add coalang alias dictionary

### PR DESCRIPTION
**Pull request by sils**

### CODE REVIEW

File: coalib/bearlib/languages/LanguageDefinition.py [hunk 1]
```diff
@@ -16,6 +23,24 @@ def __init__(self, language: str, coalang_dir=None):
...
+
+        >>> list(LanguageDefinition("cpp")['extensions'])
+        ['.c', '.cpp', '.h', '.hpp']
+
+        For some languages aliases exist, the name is case insensitive:
+
+        >>> list(LanguageDefinition("C++")['extensions'])
+        ['.c', '.cpp', '.h', '.hpp']
+
+        If no language exist, you will get a FileNotFoundError:
```
Comment (adtac) [inline coalib/bearlib/languages/LanguageDefinition.py]: backticks around `FileNotFoundError`

Comment (adtac) [inline coalib/bearlib/languages/LanguageDefinition.py]: Also I think `If no language exist` is phrased a little weirdly?

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/3003/commits/e8ca67fa13fcfd8afbb3eb3aeea170151f78439e

**Comment by sils** [comment]
reack https://github.com/coala/coala/pull/3003/commits/260342db37ff4c02c7612378edca1c04c352dd36

File: tests/bearlib/languages/LanguageDefinitionTest.py [hunk 2]
```diff
@@ -13,16 +13,6 @@ def setUp(self):
         self.section = Section("any")
         self.section.append(Setting("language", "CPP"))
 
-    def test_nonexistant_file(self):
-        self.section.append(Setting("language", "bullshit"))
-
-        with self.assertRaises(FileNotFoundError):
-            LanguageDefinition.from_section(self.section)
```
Comment (abhsag24) [inline tests/bearlib/languages/LanguageDefinitionTest.py]: why remove these?

**Comment by abhsag24** [comment]
ack 7562d24

**Comment by sils** [comment]
Covered by doctest

**Comment by sils** [comment]
ack 7562d24

File: coalib/bearlib/__init__.py [hunk 3]
```diff
@@ -11,64 +11,64 @@
...
+    >>> list(run.__metadata__.non_optional_params.keys())
+    ['new']
+    >>> list(run.__metadata__.optional_params.keys())
+    ['old']
+
+    You cannot deprecate an already deprecated setting. Don't try. It will
+    introduce non-deterministic errors in your program.
+
+    :param depr_args: A dictionary of settings as keys and their deprecated
+                      names as values.
```
Comment (abhsag24) [inline coalib/bearlib/__init__.py]: we need a bear for things like these?

**Comment by abhsag24** [comment]
ack 0bc5d05

File: tests/bearlib/languages/LanguageDefinitionTest.py (hunk 2 above)
Comment (abhsag24) [inline tests/bearlib/languages/LanguageDefinitionTest.py]: shouldn't these be two different commits?

Comment (sils) [inline tests/bearlib/languages/LanguageDefinitionTest.py]: well I added the doctest to cover our new code so it made sense to remove the redundant tests here

File: coalib/bearlib/__init__.py (hunk 3 above)
Comment (sils) [inline coalib/bearlib/__init__.py]: We could call it... IndentationBear!

**Comment by sils** [comment]
reack
1eea7f2 ConsoleInteraction: Add missing `param`
10bfef0 main: Do not print pip version
33589c9 deprecate_settings: Reindent comment

**Comment by abhsag24** [comment]
ack 6230b93

File: tests/coalaCITest.py [hunk 4]
```diff
@@ -59,6 +59,7 @@ def test_show_patch(self):
                 "-f", re.escape(filename),
                 "-b", "SpaceConsistencyTestBear",
                 "--settings", "use_spaces=True")
+            self.assertIn("Line contains ", output)  # Result message is shown
```
Comment (Uran198) [inline tests/coalaCITest.py]: This will pass regardless if the result is shown on top or not, right?
I wonder if there is a way to test that it will be on top. If there is, we should test for test.
As an idea, you can create a regexp and check for something like `<string that should be first>.*<string that should be second>.*<and so on..>`. Or, strip stuff off as you check for inclusions. If it is in separate lines, maybe you can split output linewise and check for sub-sequence

File: coalib/results/result_actions/ShowPatchAction.py [hunk 5]
```diff
@@ -84,13 +84,25 @@ def apply(self,
...
+        :param colored:
+            Whether or not to use colored output.
+        :param show_result_on_top:
+            Set this to True if you want to show the result info on top.
+            (Useful for e.g. coala_ci.)
         """
         printer = ConsolePrinter(colored)
 
+        if show_result_on_top:
+            from coalib.output.ConsoleInteraction import print_result
```
Comment (sils) [inline coalib/results/result_actions/ShowPatchAction.py]: otherwise it's a circular import, actions are used by interaction naturally

**Comment by Nosferatul** [comment]
ack e903683

File: coalib/results/result_actions/ShowPatchAction.py [hunk 6]
```diff
@@ -84,13 +84,25 @@ def apply(self,
...
+            # Most of the params are empty because they're unneeded in
+            # noninteractive mode. Yes, this cries for a refactoring...
```
Comment (sils) [inline coalib/results/result_actions/ShowPatchAction.py]: there is no issue but when the refactoring happens it'll be found easily anyway

File: tests/TestUtilities.py [hunk 7]
```diff
@@ -6,20 +6,23 @@
...
     """
     Executes the main function with the given argument string from given module.
 
-    :param function: A main function from coala_json, coala_ci module etc.
-    :param binary:   A binary to execute coala test
-    :return:         A tuple holding a return value first and
-                     a stdout output as second element.
+    :param function:    A main function from coala_json, coala_ci module etc.
+    :param binary:      A binary to execute coala test
+    :param stdout_only: Get the stdout output only, discard stderr
```
Comment (sils) [inline tests/TestUtilities.py]: well this function is used for testing only, it's not production code.

File: tests/coalaCITest.py (hunk 4 above)
Comment (sils) [inline tests/coalaCITest.py]: IMO this is sufficient, it's unlikely that there will ever be a bug that causes the result to be shown after the patch.

File: coalib/coala_main.py [hunk 8]
```diff
@@ -92,6 +92,7 @@ def run_coala(console_printer=None,
 
             if force_show_patch:
                 section['default_actions'] = "*: ShowPatchAction"
+                section['show_result_on_top'] = "yeah"
```
Comment (Nosferatul) [inline coalib/coala_main.py]: Do you want it like this?

**Comment by Nosferatul** [comment]
ack 8ca13b3

**Comment by Nosferatul** [comment]
ack f77b790

**Comment by Nosferatul** [comment]
It looks good to me 👍

**Comment by sils** [comment]
a4d89ad coala-json: Remove JSON logging functionality
dde8f0e main: Do not print pip version
81eb714 deprecate_settings: Reindent comment
6d09dbc Switch logging to stderr
14106c3 ShowPatchAction: Show result on top if needed
13566a1 ConsoleInteraction: Add missing `param`
50844d5 LanguageDefinition: Add coalang alias dictionary
reack

**Comment by sils** [comment]
@rultor merge

File: tests/TestUtilities.py (hunk 7 above)
Comment (Uran198) [inline tests/TestUtilities.py]: When testing functions get complex they should be tested ;P
